src/battle_util.c

```diff
@@ -2310,6 +2360,9 @@ u8 DoFieldEndTurnEffects(void)
               && (!(gFieldStatuses & STATUS_FIELD_TERRAIN_PERMANENT) && --gFieldTimers.psychicTerrainTimer == 0))
             {
                 gFieldStatuses &= ~(STATUS_FIELD_PSYCHIC_TERRAIN);
+                for (i = 0; i < MAX_BATTLERS_COUNT; i++)
```

Comment:
braces. It might also be advantageous to have a single function that does this loop.